prosessering-core/src/main/kotlin/no/nav/familie/prosessering/domene/TaskRepository.kt

```diff
@@ -1,6 +1,205 @@
+        WHERE t2.rn = 1
+          AND t2.opprettet_tid < :tid
+    """
+    )
+    fun findAllByStatusAndLastProcessed(status: Status, tid: LocalDateTime): List<Task>
+}
+
+/**
+ * Brukes for å generere default implementasjon for spring-jdbc [TaskRepositoryInterface]
+ */
 @Repository
-interface TaskRepository : ITaskRepostitory<ITask>
+interface GeneratedTaskRepository : TaskRepositoryInterface
+
+@Repository
+class TaskRepository(
+    private val generatedTaskRepository: GeneratedTaskRepository,
+    private val taskLoggRepository: TaskLoggRepository
+) : TaskRepositoryInterface {
+
+    override fun findByStatusInAndTriggerTidBeforeOrderByOpprettetTid(
+        status: List<Status>,
+        triggerTid: LocalDateTime,
+        page: Pageable
+    ): List<Task> {
+        return generatedTaskRepository.findByStatusInAndTriggerTidBeforeOrderByOpprettetTid(status, triggerTid, page)
+    }
+
+    override fun findByStatus(status: Status): List<Task> {
+        return generatedTaskRepository.findByStatus(status)
+    }
+
+    override fun findByStatusIn(status: List<Status>, page: Pageable): List<Task> {
+        return generatedTaskRepository.findByStatusIn(status, page)
+    }
+
+    override fun findByStatusAndTriggerTidBefore(status: Status, triggerTid: LocalDateTime): List<Task> {
+        return generatedTaskRepository.findByStatusAndTriggerTidBefore(status, triggerTid)
+    }
+
+    override fun countByStatusIn(status: List<Status>): Long {
+        return generatedTaskRepository.countByStatusIn(status)
+    }
+
+    override fun findByStatusInAndType(status: List<Status>, type: String, page: Pageable): List<Task> {
+        return generatedTaskRepository.findByStatusInAndType(status, type, page)
+    }
+
+    override fun findByPayloadAndType(payload: String, type: String): Task? {
+        return generatedTaskRepository.findByPayloadAndType(payload, type)
+    }
+
+    override fun countOpenTasks(): List<AntallÅpneTask> {
+        return generatedTaskRepository.countOpenTasks()
+    }
+
+    override fun findAllByStatusAndLastProcessed(status: Status, tid: LocalDateTime): List<Task> {
+        return generatedTaskRepository.findAllByStatusAndLastProcessed(status, tid)
+    }
+
+    @Transactional
+    override fun <S : Task> save(entity: S): S {
+        val task = generatedTaskRepository.save(entity)
+        validerTask(entity)
+        if (entity.versjon == 0L) {
+            taskLoggRepository.save(TaskLogg(type = Loggtype.UBEHANDLET, taskId = task.id))
+        } else if (entity.logg != null) {
+            validerTaskLogg(entity.logg)
+            taskLoggRepository.save(entity.logg.copy(taskId = task.id))
+        }
+        @Suppress("UNCHECKED_CAST")
+        return task.copy(logg = null) as S
+    }
+
+    private fun validerTaskLogg(logg: TaskLogg) {
+        if (logg.id != 0L || logg.taskId != 0L) {
+            error("Kan ikke sette id'er på logg")
+        }
+    }
+
+    private fun <S : Task> validerTask(entity: S) {
```

Comment:
Task er final, så S mmå være Task     private fun validerTask(entity: Task) {
```suggestion
    private fun <S : Task> validerTask(entity: S) {
```

Reply:
Pga interfacet så går ikke det `<S extends T> S save(S entity);`

Reply:
Fungerte godt nok til at alle testene kjører som de skal.

Reply:
Ah jeg såg feil på denne, trodde det var en av de metodene som var overridet, denne burde selvfølgelig ikke ha `<S: Task>`. Skal fikse 👍